kore/app/exec/Main.hs

```diff
@@ -306,54 +308,73 @@ mainWithOptions
                     { SMT.timeOut = smtTimeOut
                     , SMT.preludeFile = smtPrelude
                     }
-        parsedDefinition <- parseDefinition definitionFileName
-        indexedDefinition@(indexedModules, _) <-
-            verifyDefinitionWithBase
-                Nothing
-                True
-                parsedDefinition
-        indexedModule <- mainModule mainModuleName indexedModules
-        searchParameters <-
-            case koreSearchOptions of
-                Nothing -> return Nothing
-                Just KoreSearchOptions { searchFileName, bound, searchType } ->
-                    do
-                        searchPattern <-
-                            mainParseSearchPattern indexedModule searchFileName
-                        let searchConfig = Search.Config { bound, searchType }
-                        (return . Just) (searchPattern, searchConfig)
-        proveParameters <-
-            case koreProveOptions of
+        withLogger koreLogOptions (\logger -> do
+            parsedDefinition <-
+                runReaderT
```

Comment:
Instead of wrapping each individual call in `runReaderT`, why not put one `runReaderT` at the top of the block?